examples/inspection/plot_permutation_importance.py

```diff
@@ -172,3 +172,33 @@
 ax.set_title("Permutation Importances (train set)")
 fig.tight_layout()
 plt.show()
+
+# %%
```

Comment:
IRL with @ogrisel, he was proposing to put this example under the MDI computation.
Additionally, we should mention that MDI and permutation on OOB are only computed on the feature space provided to the forest (i.e. after the preprocessing) and that the last function allows inspecting at different stages.

In addition, we should make it explicit that MDI is computed from the in-sample and would suffer from a bias if the model learnt overfits. The permutation OOB offers an alternative similar to the external permutation importance. It is even better if we have limited data since we are not required to split the data into training/testing and leverage the OOB.

Finally, we should mention that the estimate of the variance of the permutation importance could be more efficient with the OOB scheme: `n_repeats` would be required to be set to `n_estimators` to have as many estimates.